indico/modules/events/timetable/templates/pdf/_program.html

```diff
@@ -0,0 +1,324 @@
+    </section>
+    {% if show_contrib_content %}
+        <div class="session-block-content">
+            {{ _render_description(contrib.description) }}
+            {% if subcontribs %}
+                {{ _render_subcontribs(contrib, config, show_children_location) }}
+            {% endif %}
+        </div>
+    {% endif %}
+{% endmacro %}
+
+{% macro _render_session_block(entry, config, program_config, show_children_location) %}
+    {% set session_block = entry.object %}
+    {% set sess = session_block.session %}
+    {% set conveners = session_block.person_links %}
+    {% set contributions = session_block.contributions %}
+    {% set show_session_block_content = entry.children or (config.show_session_description and sess.description) %}
+
+    <section
+        id="entry-{{ entry.id }}"
+        class="session-block-title inherited-bg-color inherited-txt-color {{ 'only-round-top' if show_session_block_content else ''}}"
+    >
+        {{ _render_entry_header(entry, config, program_config) }}
+    </section>
+    {% if show_session_block_content %}
+        <div class="session-block-content">
+            {% if config.show_session_description and sess.description %}
+                {{ _render_description(sess.description) }}
+            {% endif %}
+            {% if entry.children %}
+                {{ _render_block_child_entries(sess, entry.children, config, show_children_location) }}
+            {% endif %}
+        </div>
+    {% endif %}
+{% endmacro %}
+
+{% macro _render_block_child_entries(sess, children, config, show_children_location) %}
+    <div class="session-block-content-box inherited-border-color">
+        <ul class="contrib-timeline">
+            {% for child in children %}
+                <li class="inherited-border-color">
+                    <div class="contrib-heading">
+                        {% if not sess.is_poster %}
+                            <span class="timebox inherited-bg-color inherited-txt-color">
+                                {{ format_interval(child.start_dt.astimezone(child.event.tzinfo), child.end_dt.astimezone(child.event.tzinfo), format='HHmm') }}
+                                {% if config.show_length_contribs %}
+                                    ({{ (child.end_dt - child.start_dt)|format_human_timedelta(narrow=True) }})
+                                {% endif %}
+                            </span>
+                        {% endif %}
+                        <h4>{{ child.object.title }}</h4>
+                    </div>
+                    {% if child.type.name == 'CONTRIBUTION' %}
+                        <div class="contrib-timeline-content" >
+                            {{ _render_child_contrib(child.object, config, show_children_location) }}
+                        </div>
+                    {% endif %}
+                </li>
+            {% endfor %}
+        </ul>
+    </div>
+{% endmacro %}
+
+{% macro _render_child_contrib(contrib, config, show_location) %}
```

Comment:
I would add a comment here mentioning that `contrib` can be a `Contribution` or a `SubContribution`. This is not clear from just looking at the code and with the naming I would not expect a SubContribution...